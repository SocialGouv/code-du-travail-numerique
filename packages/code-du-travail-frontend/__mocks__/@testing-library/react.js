/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
