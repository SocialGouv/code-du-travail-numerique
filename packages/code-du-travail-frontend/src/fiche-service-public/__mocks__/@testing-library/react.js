import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
