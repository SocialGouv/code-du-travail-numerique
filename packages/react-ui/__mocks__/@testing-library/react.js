import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
