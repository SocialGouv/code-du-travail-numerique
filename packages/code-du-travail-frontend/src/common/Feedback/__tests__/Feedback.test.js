import { render } from "@testing-library/react";
import React from "react";

import { Feedback } from "..";

global.fetch = jest.fn().mockResolvedValue({ json: () => ({ error: false }) });

jest.mock("@socialgouv/matomo-next", () => {
  let events = [];
  return {
    events,
    flushEvents() {
      events = [];
    },
    push: (event) => events.push(event),
  };
});

const { events, flushEvents } = require("@socialgouv/matomo-next");

afterEach(flushEvents);

describe("<Feedback/>", () => {
  it("should render", () => {
    const { container } = render(
      <Feedback url="http://server/fiches/fiche1" />,
    );
    expect(container).toMatchSnapshot();
  });
  it("should render form once user answer yes", () => {
    const { container, getByText } = render(
      <Feedback url="http://server/fiches/fiche1" />,
    );
    const button = getByText(/oui/i);
    button.click();
    expect(container).toMatchSnapshot();
  });
  it("should render form once user answer no", () => {
    const { container, getByText } = render(
      <Feedback url="http://server/fiches/fiche1" />,
    );
    const button = getByText(/non/i);
    button.click();
    expect(container).toMatchSnapshot();
  });
  it("should send matomo event when click yes", () => {
    const { getByText } = render(
      <Feedback url="http://server/fiches/fiche1" />,
    );
    const button = getByText(/oui/i);
    button.click();
    expect(JSON.stringify(events)).toMatchSnapshot();
    flushEvents();
  });
  it("should send matomo event when click no", () => {
    const { getByText } = render(
      <Feedback url="http://server/fiches/fiche1" />,
    );
    const button = getByText(/non/i);
    button.click();
    expect(JSON.stringify(events)).toMatchSnapshot();
  });
});
