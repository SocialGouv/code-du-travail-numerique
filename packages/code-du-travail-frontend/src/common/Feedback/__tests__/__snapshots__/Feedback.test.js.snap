// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feedback/> should render 1`] = `
.c2 {
  max-width: 124rem;
  margin: 0 auto;
  padding: 0 2rem;
}

.c0 {
  padding: 3.2rem 0;
  background-color: transparent;
  color: #3e486e;
}

.c3 {
  padding: 2rem 2.4rem;
  color: #3e486e;
  border-radius: 0.6rem;
  border: 1px solid #bbcadf;
  background-color: #fff;
}

.c3 > *:first-child {
  margin-top: 0;
}

.c3 > *:last-child {
  margin-bottom: 0;
}

.c7 {
  position: absolute;
  background-color: #f66663;
  border-radius: 0;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c6 {
  position: relative;
  display: block;
  margin: 3.2rem 0 2rem 0;
  color: #2f3b6c;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
  margin-left: -2.4rem;
  padding-left: 2.4rem;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c10:link,
.c10:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c10:not([disabled]):hover,
.c10:not([disabled]):active,
.c10:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c10[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c8 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
  margin: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

.c5 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
}

.c11 + .c11 {
  margin-left: 4rem;
}

@media (max-width:600px) {
  .c2 {
    padding: 0 1rem;
  }
}

@media print {
  .c2 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c0 {
    padding: 2.4rem 0;
  }
}

@media (max-width:600px) {
  .c3 {
    padding: 1.6rem 1rem;
  }
}

@media print {
  .c3 {
    padding: 0 5pt;
    border: none;
  }
}

@media (max-width:600px) {
  .c6 {
    margin-left: -1rem;
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c6 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c10 {
    font-size: 1.4rem;
  }
}

@media print {
  .c10 {
    display: none;
  }
}

@media (max-width:600px) {
  .c10 {
    padding: 0 3rem;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:1180px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    margin-right: auto;
    margin-left: auto;
  }
}

@media (max-width:1180px) {
  .c5 {
    margin-bottom: 1.6rem;
  }
}

@media (max-width:600px) {
  .c5 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media (max-width:1180px) {
  .c9 {
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (max-width:1180px) {
  .c11 + .c11 {
    margin-left: 0;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <strong
            class="c5 c6"
          >
            <span
              class="c7"
            />
            <p
              aria-level="2"
              class="c8"
              role="heading"
            >
              Avez-vous trouvé la réponse à votre question ?
            </p>
          </strong>
          <p
            class="c9"
          >
            <button
              class="c10 c11 "
            >
              Non
            </button>
            <button
              class="c10 c11 "
            >
              Oui
            </button>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Feedback/> should render form once user answer no 1`] = `
.c2 {
  max-width: 124rem;
  margin: 0 auto;
  padding: 0 2rem;
}

.c0 {
  padding: 3.2rem 0;
  background-color: transparent;
  color: #3e486e;
}

.c3 {
  padding: 2rem 2.4rem;
  color: #3e486e;
  border-radius: 0.6rem;
  border: 1px solid #bbcadf;
  background-color: #fff;
}

.c3 > *:first-child {
  margin-top: 0;
}

.c3 > *:last-child {
  margin-bottom: 0;
}

.c7 {
  position: absolute;
  background-color: #f66663;
  border-radius: 0;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c6 {
  position: relative;
  display: block;
  margin: 3.2rem 0 2rem 0;
  color: #2f3b6c;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
  margin-left: -2.4rem;
  padding-left: 2.4rem;
}

.c20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 4rem;
  padding: 0 3rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c20:link,
.c20:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c20:not([disabled]):hover,
.c20:not([disabled]):active,
.c20:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c20[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c22 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  color: #f66663;
  font-weight: 600;
  font-size: 1.6rem;
  line-height: 1.625;
  vertical-align: baseline;
  text-align: left;
  background: none;
  border: none;
  border-radius: 0;
  overflow: visible;
}

.c22:focus,
.c22:hover,
.c22:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c24 {
  width: 2.6rem;
  height: 1.4rem;
  margin: 0 0.4rem 0 1rem;
  -webkit-transition: -webkit-transform 250ms linear;
  -webkit-transition: transform 250ms linear;
  transition: transform 250ms linear;
}

.c19:hover .c23 {
  -webkit-transform: translateX(4px);
  -ms-transform: translateX(4px);
  transform: translateX(4px);
}

.c17 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  padding: 0 !important;
  overflow: hidden !important;
  white-space: nowrap !important;
  border: 0 !important;
  -webkit-clip: rect(0,0,0,0) !important;
  clip: rect(0,0,0,0) !important;
}

.c16 {
  color: #9298af;
  font-size: 1.2rem;
  font-weight: 400;
}

.c8 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  cursor: pointer;
}

.c12 {
  position: relative;
  display: block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: calc(1.6rem * 1.25);
  height: calc(1.6rem * 1.25);
  margin: calc(1.6rem / 5) calc(2 * 1.6rem / 3) 0 0;
  background: #fff;
  border: 1px solid #bbcadf;
  border-radius: 0.4rem;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c12:before {
  position: absolute;
  top: 50%;
  left: 50%;
  display: none;
  width: calc(1.6rem / 3);
  height: calc(1.6rem / 1.7);
  border: 1px solid #fff;
  border-width: 0 calc(1.6rem / 10) calc(1.6rem / 10) 0;
  -webkit-transform: translate(-50%,-60%) rotate(45deg);
  -ms-transform: translate(-50%,-60%) rotate(45deg);
  transform: translate(-50%,-60%) rotate(45deg);
  content: "";
}

.c12:checked {
  background-color: #f66663;
}

.c12:checked:before {
  display: block;
}

.c14 {
  width: 100%;
  min-height: 12rem;
  padding: 2rem;
  color: #3e486e;
  font-size: 1.6rem;
  font-family: "Open Sans",sans-serif;
  line-height: inherit;
  background: #fff;
  border: 1px solid transparent;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c14::-webkit-input-placeholder {
  color: #9298af;
}

.c14::-moz-placeholder {
  color: #9298af;
}

.c14:-ms-input-placeholder {
  color: #9298af;
}

.c14::placeholder {
  color: #9298af;
}

.c14::-webkit-outer-spin-button,
.c14::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c14:focus {
  border-color: #7994d4;
}

.c14:focus::-webkit-input-placeholder {
  color: transparent;
}

.c14:focus::-moz-placeholder {
  color: transparent;
}

.c14:focus:-ms-input-placeholder {
  color: transparent;
}

.c14:focus::placeholder {
  color: transparent;
}

.c9 {
  border: none;
  padding: 0;
}

.c10 {
  font-weight: 600;
  padding: 1rem 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

.c5 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.c21 + .c21 {
  margin-left: 4rem;
}

.c15 {
  max-width: 100%;
}

.c13 {
  font-weight: 600;
  display: block;
  margin: 2rem 0 1rem 0;
}

.c18 {
  margin: 3.2rem 0;
}

@media (max-width:600px) {
  .c2 {
    padding: 0 1rem;
  }
}

@media print {
  .c2 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c0 {
    padding: 2.4rem 0;
  }
}

@media (max-width:600px) {
  .c3 {
    padding: 1.6rem 1rem;
  }
}

@media print {
  .c3 {
    padding: 0 5pt;
    border: none;
  }
}

@media (max-width:600px) {
  .c6 {
    margin-left: -1rem;
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c6 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {

}

@media print {

}

@media (max-width:600px) {

}

@media (max-width:600px) {
  .c20 {
    font-size: 1.4rem;
  }
}

@media print {
  .c20 {
    display: none;
  }
}

@media (max-width:600px) {
  .c22 {
    font-size: 1.4rem;
  }
}

@media print {
  .c22 {
    display: none;
  }
}

@media (max-width:600px) {
  .c14 {
    width: 100%;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:1180px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    margin-right: auto;
    margin-left: auto;
  }
}

@media (max-width:1180px) {
  .c5 {
    margin-bottom: 1.6rem;
  }
}

@media (max-width:600px) {
  .c5 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media (max-width:1180px) {

}

@media (max-width:1180px) {
  .c21 + .c21 {
    margin-left: 0;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <strong
            class="c5 c6"
          >
            <span
              class="c7"
            />
            <p
              class="c8"
              role="status"
            >
              Merci pour votre réponse.
            </p>
          </strong>
        </div>
        <form>
          <fieldset
            class="c9"
          >
            <legend
              class="c10"
            >
              Pouvez-vous nous en dire plus ?
            </legend>
            <label
              class="c11"
              for="unclear"
            >
              <input
                class="c12"
                id="unclear"
                name="category"
                type="checkbox"
                value="unclear"
              />
              Les informations ne sont pas claires.
            </label>
            <label
              class="c11"
              for="unrelated"
            >
              <input
                class="c12"
                id="unrelated"
                name="category"
                type="checkbox"
                value="unrelated"
              />
              Cette page ne correspond pas à ma recherche ou à ma situation.
            </label>
            <label
              class="c11"
              for="unsatisfied"
            >
              <input
                class="c12"
                id="unsatisfied"
                name="category"
                type="checkbox"
                value="unsatisfied"
              />
              Je ne suis pas satisfait de cette réglementation.
            </label>
            <label
              class="c11"
              for="wrong"
            >
              <input
                class="c12"
                id="wrong"
                name="category"
                type="checkbox"
                value="wrong"
              />
              Les informations me semblent fausses.
            </label>
          </fieldset>
          <label
            class="c13"
            for="suggestion"
          >
            Faire une suggestion pour améliorer cette page
          </label>
          <textarea
            class="c14 c15"
            cols="50"
            id="suggestion"
            maxlength="150"
            name="suggestion"
            placeholder="Une idée, une suggestion..."
            rows="3"
          />
          <div>
            <span
              class="c16"
            >
              150
               caractères restants
            </span>
            <div
              class="c17"
              role="status"
            >
              150
               caractères restants
            </div>
          </div>
          <div
            class="c18"
          >
            <button
              class="c19 c20 c21 "
            >
              Envoyer
            </button>
          </div>
        </form>
        <p>
          Pour obtenir une réponse à votre question de droit du travail, nous vous invitons à joindre
           
          <button
            class="c19 c22"
            type="button"
          >
            les services du ministère du Travail en région
            <svg
              aria-hidden="true"
              class="c23 c24"
              fill="none"
              viewBox="0 0 28 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
                fill="currentColor"
              />
            </svg>
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Feedback/> should render form once user answer yes 1`] = `
.c2 {
  max-width: 124rem;
  margin: 0 auto;
  padding: 0 2rem;
}

.c0 {
  padding: 3.2rem 0;
  background-color: transparent;
  color: #3e486e;
}

.c3 {
  padding: 2rem 2.4rem;
  color: #3e486e;
  border-radius: 0.6rem;
  border: 1px solid #bbcadf;
  background-color: #fff;
}

.c3 > *:first-child {
  margin-top: 0;
}

.c3 > *:last-child {
  margin-bottom: 0;
}

.c7 {
  position: absolute;
  background-color: #f66663;
  border-radius: 0;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c6 {
  position: relative;
  display: block;
  margin: 3.2rem 0 2rem 0;
  color: #2f3b6c;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
  margin-left: -2.4rem;
  padding-left: 2.4rem;
}

.c15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 4rem;
  padding: 0 3rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c15:link,
.c15:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c15:not([disabled]):hover,
.c15:not([disabled]):active,
.c15:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c15[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c13 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  padding: 0 !important;
  overflow: hidden !important;
  white-space: nowrap !important;
  border: 0 !important;
  -webkit-clip: rect(0,0,0,0) !important;
  clip: rect(0,0,0,0) !important;
}

.c12 {
  color: #9298af;
  font-size: 1.2rem;
  font-weight: 400;
}

.c8 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c10 {
  width: 100%;
  min-height: 12rem;
  padding: 2rem;
  color: #3e486e;
  font-size: 1.6rem;
  font-family: "Open Sans",sans-serif;
  line-height: inherit;
  background: #fff;
  border: 1px solid transparent;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c10::-webkit-input-placeholder {
  color: #9298af;
}

.c10::-moz-placeholder {
  color: #9298af;
}

.c10:-ms-input-placeholder {
  color: #9298af;
}

.c10::placeholder {
  color: #9298af;
}

.c10::-webkit-outer-spin-button,
.c10::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c10:focus {
  border-color: #7994d4;
}

.c10:focus::-webkit-input-placeholder {
  color: transparent;
}

.c10:focus::-moz-placeholder {
  color: transparent;
}

.c10:focus:-ms-input-placeholder {
  color: transparent;
}

.c10:focus::placeholder {
  color: transparent;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

.c5 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.c16 + .c16 {
  margin-left: 4rem;
}

.c11 {
  max-width: 100%;
}

.c9 {
  font-weight: 600;
  display: block;
  margin: 2rem 0 1rem 0;
}

.c14 {
  margin: 3.2rem 0;
}

@media (max-width:600px) {
  .c2 {
    padding: 0 1rem;
  }
}

@media print {
  .c2 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c0 {
    padding: 2.4rem 0;
  }
}

@media (max-width:600px) {
  .c3 {
    padding: 1.6rem 1rem;
  }
}

@media print {
  .c3 {
    padding: 0 5pt;
    border: none;
  }
}

@media (max-width:600px) {
  .c6 {
    margin-left: -1rem;
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c6 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {

}

@media print {

}

@media (max-width:600px) {

}

@media (max-width:600px) {
  .c15 {
    font-size: 1.4rem;
  }
}

@media print {
  .c15 {
    display: none;
  }
}

@media (max-width:600px) {
  .c10 {
    width: 100%;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:1180px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    margin-right: auto;
    margin-left: auto;
  }
}

@media (max-width:1180px) {
  .c5 {
    margin-bottom: 1.6rem;
  }
}

@media (max-width:600px) {
  .c5 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media (max-width:1180px) {

}

@media (max-width:1180px) {
  .c16 + .c16 {
    margin-left: 0;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <strong
            class="c5 c6"
          >
            <span
              class="c7"
            />
            <p
              class="c8"
              role="status"
            >
              Merci pour votre réponse.
            </p>
          </strong>
        </div>
        <form>
          <label
            class="c9"
            for="suggestion"
          >
            Faire une suggestion pour améliorer cette page
          </label>
          <textarea
            class="c10 c11"
            cols="50"
            id="suggestion"
            maxlength="150"
            name="suggestion"
            placeholder="Une idée, une suggestion..."
            rows="3"
          />
          <div>
            <span
              class="c12"
            >
              150
               caractères restants
            </span>
            <div
              class="c13"
              role="status"
            >
              150
               caractères restants
            </div>
          </div>
          <div
            class="c14"
          >
            <button
              class="c15 c16 "
            >
              Envoyer
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Feedback/> should send piwik event when click no 1`] = `"[]"`;

exports[`<Feedback/> should send piwik event when click yes 1`] = `"[]"`;
